src/sys/boutexception.cxx

```diff
@@ -82,27 +92,21 @@ std::string BoutException::getBacktrace() const {
     FILE* file = popen(syscom.c_str(), "r");
     if (file != nullptr) {
       std::array<char, 1024> out{};
-      char* retstr = nullptr;
+      const char* retstr = nullptr;
       std::string buf;
       while ((retstr = fgets(out.data(), out.size() - 1, file)) != nullptr) {
         buf += retstr;
       }
       int const status = pclose(file);
       if (status == 0) {
-        backtrace_message += buf;
+        backtrace_message = fmt::format("{}{}", backtrace_message, buf);
       }
     }
   }
 #else
   backtrace_message = "Stacktrace not enabled.\n";
 #endif
 
-  return backtrace_message + msg_stack.getDump() + "\n" + header + message + "\n";
-}
-
-void BoutException::makeBacktrace() {
```

Comment:
This is a breaking ABI change. Should we just keep it as a no-op?

Reply:
No, I don't think that's necessary, I'm not sure it ever really made sense to make this part of the public interface as we always called it in the constructor anyway.

Also, next release will have more meaningful breaking changes :)